ultralytics/yolo/engine/predictor.py

```diff
@@ -220,6 +217,51 @@ def predict_cli(self, source=None, model=None, return_outputs=False):
         for _ in (self.__call__(source, model, return_outputs)):
             pass
 
+    def setup_model(self, model):
+        device = select_device(self.args.device)
+        model = model or self.args.model
+        self.args.half &= device.type != 'cpu'  # half precision only supported on CUDA
+        model = AutoBackend(model, device=device, dnn=self.args.dnn, fp16=self.args.half)
+        self.model = model
+        self.device = device
+        self.model.eval()
+        return model.stride, model.pt
+
+    def inference(self, img):
+        # minimal inference demo for python interface
+        # supporting PIL/ndarray/tensor
+        if not isinstance(img, list):
+            img = [img]
+        img = [self._single_check(im) for im in img]
+        auto = all(x.shape == img[0].shape for x in img) and self.model.pt
+        im = [self._single_preprocess(im, self.model.stride, auto) for im in img]
+        im = np.stack(im, 0) if len(im) > 1 else im[0]
+
+        im = self.preprocess(im)
+        if len(im.shape) == 3:
+            im = im[None]  # expand for batch dim
+        preds = self.model(im)
+        preds = self.postprocess(preds, im, img)
+        return preds
```

Comment:
@AyushExel ideally once your `Results` class is done, you can just update this line to:
```python
return Result(preds)
```

Reply:
yup that sounds good!